awswrangler/neptune/client.py

```diff
@@ -0,0 +1,205 @@
+                      "config-file, ec2-credentials-file, boto-config, container-role, iam-role\n")
+                print("Go to https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html for more "
+                      "details on configuring your IAM credentials.")
+                return req
+            SigV4Auth(frozen_creds, service, self.region).add_auth(req)
+            prepared_iam_req = req.prepare()
+            return prepared_iam_req
+        else:
+            return req
+
+
+    def read_opencypher(self, query: str, headers:Dict[str, Any] = None) -> Dict[str, Any]:
+        if headers is None:
+            headers = {}
+
+        if 'content-type' not in headers:
+            headers['content-type'] = 'application/x-www-form-urlencoded'
+
+        url = f'{self._http_protocol}://{self.host}:{self.port}/openCypher'
+        data = {
+            'query': query
+        }
+
+        req = self._prepare_request('POST', url, data=data, headers=headers)
+        res = self._http_session.send(req)
+
+        return res.json()
+
+    
+    def read_gremlin(self, query, headers:Dict[str, Any] = None) -> Dict[str, Any]:
+        try:
+            nest_asyncio.apply()
+            uri = f'{self._http_protocol}://{self.host}:{self.port}/gremlin'
+            request = self._prepare_request('GET', uri)
+            ws_url = f'{self._ws_protocol}://{self.host}:{self.port}/gremlin'
+            c = client.Client(ws_url, 'g', headers=dict(request.headers))
+            result = c.submit(query)
+            future_results = result.all()
+            results = future_results.result()
+            c.close()
+            return self._gremlin_results_to_dict(results)
+        except Exception as e:
+            c.close()
+            raise e
+
+    
+    def read_sparql(self, query, headers:Dict[str, Any] = None) -> Dict[str, Any]:
+        if headers is None:
+            headers = {}
+        
+        data = {'query': query}
+
+        if 'content-type' not in headers:
+            headers['content-type'] = 'application/x-www-form-urlencoded'
+
+        uri = f'{self._http_protocol}://{self.host}:{self.port}/sparql'
+        req = self._prepare_request('POST', uri, data=data, headers=headers)
+        res = self._http_session.send(req)
+        return res
+
+
+    def status(self):
+        url = f'{self._http_protocol}://{self.host}:{self.port}/status'
+        req = self._prepare_request('GET', url, data='')
+        res = self._http_session.send(req)
+        if res.status_code == 200:
+            return res.json()
+        else:
+            _logger.error("Error connecting to Amazon Neptune cluster. Please verify your connection details")
+            raise ConnectionError(res.status_code)
+
+
+    def _gremlin_results_to_dict(self, result) -> Dict[str, Any]:
+        res=[]
+
+        # For lists or paths unwind them
+        if isinstance(result, list) or isinstance(result, Path):
+            for x in result:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)